File: classes/Paciente.php
<?php
// classes/Paciente.php

// Definição da classe Paciente
class Paciente
{
    // Declaração de propriedades privadas da classe
    private $id, $nome, $dataNascimento, $telefone, $email, $nomeMae, $idade, $medicamento, $patologia;

    // Construtor da classe para inicializar as propriedades
    public function __construct($id, $nome, $dataNascimento, $telefone, $email, $nomeMae, $idade, $medicamento, $patologia)
    {
        $this->id = $id; // ID do paciente
        $this->nome = $nome; // Nome completo do paciente
        $this->dataNascimento = $dataNascimento; // Data de nascimento do paciente
        $this->telefone = $telefone; // Telefone do paciente
        $this->email = $email; // E-mail do paciente
        $this->nomeMae = $nomeMae; // Nome da mãe do paciente
        $this->idade = $idade; // Idade do paciente
        $this->medicamento = $medicamento; // Medicamento associado ao paciente
        $this->patologia = $patologia; // Patologia associada ao paciente
    }

    // Método para obter o ID do paciente
    public function getId()
    {
        return $this->id;
    }

    // Método para salvar os dados do paciente no banco de dados
    public function salvar($conn)
    {
        // Verifica se o e-mail já existe no banco de dados
        if (!empty($this->email)) {
            $verificaEmail = "SELECT COUNT(*) as total FROM pacientes WHERE email = ?"; // Consulta SQL para verificar duplicidade de e-mail
            $stmt = $conn->prepare($verificaEmail); // Prepara a consulta
            $stmt->bind_param("s", $this->email); // Substitui o placeholder pelo e-mail do paciente
            $stmt->execute(); // Executa a consulta
            $result = $stmt->get_result(); // Obtém o resultado da consulta
            $dados = $result->fetch_assoc(); // Extrai os dados do resultado

            // Se o e-mail já estiver cadastrado, lança uma exceção
            if ($dados['total'] > 0) {
                throw new Exception("Este e-mail já está cadastrado no sistema.");
            }
        }

        // Consulta SQL para inserir os dados do paciente no banco de dados
        $sql = "INSERT INTO pacientes (
            idPaciente, nomeCompleto, dataNascimento, idade, telefone, email, nomeMae, nomeMedicamento, nomePatologia
        ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)";
        
        $stmt = $conn->prepare($sql); // Prepara a consulta SQL para inserção
        
        // Define valores padrão para medicamento e patologia caso estejam nulos
        $medicamento = $this->medicamento ?? '';
        $patologia = $this->patologia ?? '';
        
        // Substitui os placeholders pelos valores das propriedades do objeto
        $stmt->bind_param("sssisssss", 
            $this->id, // ID do paciente
            $this->nome, // Nome completo do paciente
            $this->dataNascimento, // Data de nascimento do paciente
            $this->idade, // Idade do paciente
            $this->telefone, // Telefone do paciente
            $this->email, // E-mail do paciente
            $this->nomeMae, // Nome da mãe do paciente
            $medicamento, // Medicamento associado ao paciente
            $patologia // Patologia associada ao paciente
        );
        
        $stmt->execute(); // Executa a consulta para salvar os dados no banco de dados
    }
}

File: classes/java/Paciente.java
package java;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class Paciente {
    private String id;
    private String nome;
    private String dataNascimento;
    private String telefone;
    private String email;
    private String nomeMae;
    private int idade;
    private String medicamento;
    private String patologia;

    public Paciente(String id, String nome, String dataNascimento, String telefone,
                    String email, String nomeMae, int idade, String medicamento, String patologia) {
        this.id = id;
        this.nome = nome;
        this.dataNascimento = dataNascimento;
        this.telefone = telefone;
        this.email = email;
        this.nomeMae = nomeMae;
        this.idade = idade;
        this.medicamento = medicamento;
        this.patologia = patologia;
    }

    public String getId() {
        return id;
    }

    public void salvar(Connection conn) throws SQLException {
        // Verifica se o e-mail já existe no banco
        if (email != null && !email.isEmpty()) {
            String verificaEmail = "SELECT COUNT(*) AS total FROM pacientes WHERE email = ?";
            try (PreparedStatement stmt = conn.prepareStatement(verificaEmail)) {
                stmt.setString(1, email);
                try (ResultSet rs = stmt.executeQuery()) {
                    if (rs.next() && rs.getInt("total") > 0) {
                        throw new SQLException("Este e-mail já está cadastrado no sistema.");
                    }
                }
            }
        }

        // Insere os dados do paciente
        String sql = "INSERT INTO pacientes (idPaciente, nomeCompleto, dataNascimento, idade, telefone, email, nomeMae, nomeMedicamento, nomePatologia) " +
                     "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)";

        try (PreparedStatement stmt = conn.prepareStatement(sql)) {
            stmt.setString(1, id);
            stmt.setString(2, nome);
            stmt.setString(3, dataNascimento);
            stmt.setInt(4, idade);
            stmt.setString(5, telefone);
            stmt.setString(6, email);
            stmt.setString(7, nomeMae);
            stmt.setString(8, medicamento != null ? medicamento : "");
            stmt.setString(9, patologia != null ? patologia : "");

            stmt.executeUpdate();
        }
    }

    public static void listar(Connection conn) {
        // TODO Auto-generated method stub
        throw new UnsupportedOperationException("Unimplemented method 'listar'");
    }

    public static Paciente buscarPorId(Connection conn, String idPaciente) {
        // TODO Auto-generated method stub
        throw new UnsupportedOperationException("Unimplemented method 'buscarPorId'");
    }
}
